### PR TITLE
chore: P4PU-145 create feign client for biz-events service

### PR DESCRIPTION
**Pull request by Giuseppe-LaManna**
#### List of Changes
- Added Feign Client for BizEvents Api transactionsList
- Added Junit test

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by and-mora** [comment]
according to the recent handbook and versioning this PR can be marked with the semantic commit `chore` since the code is not reachable.

**Comment by and-mora** [comment]
fix the sonar issues, please

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -17,4 +17,8 @@ management:
     web:
       exposure.include: info, health
 
-apiKey.bizEvents: ${BIZ_EVENTS_SERVICE_API_KEY:}
\ No newline at end of file
+apiKey.biz-events: \${BIZ_EVENTS_SERVICE_API_KEY:}
+
+rest-client:
+  biz-events:
+    baseUrl: \${BIZ_EVENTS_BASE_URL:}
```
Comment (and-mora) [inline src/main/resources/application.yml:24]: i think we can group the properties related to biz-events under the same root property

File: src/main/java/it/gov/pagopa/arc/connector/bizevents/BizEventsConnector.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package it.gov.pagopa.arc.connector.bizevents;
+
+import it.gov.pagopa.arc.connector.bizevents.dto.BizEventsTransactionsListDTO;
+
+public interface BizEventsConnector {
+    BizEventsTransactionsListDTO transactionsList(String fiscalCode, String continuationToken, int size);
```
Comment (and-mora) [inline src/main/java/it/gov/pagopa/arc/connector/bizevents/BizEventsConnector.java:6]: may you change the method name into something verbish like 'getSomething'?

File: src/main/java/it/gov/pagopa/arc/connector/bizevents/BizEventsConnectorImpl.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        BizEventsTransactionsListDTO bizEventsTransactionsListDTO;
+        try {
+            bizEventsTransactionsListDTO = bizEventsRestClient.transactionsList(apikey, fiscalCode, continuationToken, size);
+        }catch (FeignException e) {
+            bizEventsTransactionsListDTO =
+                    BizEventsTransactionsListDTO
+                            .builder()
+                            .transactions(new ArrayList<>())
+                            .build();
+        }
```
Comment (and-mora) [inline src/main/java/it/gov/pagopa/arc/connector/bizevents/BizEventsConnectorImpl.java:34]: we lack logging (and related observability) here.

File: src/test/resources/stub/mappings/bizEventsService/bizEvents_Get_TransactionsList_OK.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+          "payeeName": "Comune di Milano",
+          "payeeTaxCode": "MI_XXX",
+          "amount": "180,00",
+          "transactionDate": "2024-03-27T13:07:25Z",
+          "isCart": false,
+          "isPayer": true,
+          "isDebtor": true
+        }
+      ]
+    }
```
Comment (and-mora) [inline src/test/resources/stub/mappings/bizEventsService/bizEvents_Get_TransactionsList_OK.json:40]: i remember bizEvents returns an array (not an object with a array property), am i wrong?

File: src/test/resources/stub/mappings/bizEventsService/bizEvents_Get_TransactionsList_DEFAULT.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
+{
+  "request": {
+    "method": "GET",
+    "urlPathPattern": "/bizEventsMock/transactions",
+    "headers": {
+      "x-api-key": {
+        "matches": "x_api_key([0-9]?)+"
+      },
+      "x-fiscal-code": {
+        "matches": "DUMMY_FISCAL_CODE_DEFAULT"
```
Comment (and-mora) [inline src/test/resources/stub/mappings/bizEventsService/bizEvents_Get_TransactionsList_DEFAULT.json:10]: we might call it `ERROR` instead of `DEFAULT`

**Review by and-mora — CHANGES_REQUESTED** [review]

File: src/test/resources/stub/mappings/bizEventsService/bizEvents_Get_TransactionsList_OK.json (hunk 4 above)
Comment (Giuseppe-LaManna) [reply src/test/resources/stub/mappings/bizEventsService/bizEvents_Get_TransactionsList_OK.json:40]: yes that's wrong, the biz-events api returns an object with the transactions property inside which is an array of transactions.

File: src/main/java/it/gov/pagopa/arc/connector/bizevents/BizEventsConnectorImpl.java (hunk 3 above)
Comment (and-mora) [reply src/main/java/it/gov/pagopa/arc/connector/bizevents/BizEventsConnectorImpl.java:34]: plus:  I don't like very much the "catch all status codes" mode. We could change to "catch the 404 or rethrow the exception", I find it more transparent as if any errors occurs we can see it. 
Later we can think to catch the other status codes when we know what to do

File: helm/values.yaml [hunk 6]
```diff
@@ -61,6 +61,9 @@ microservice-chart:
     APPLICATIONINSIGHTS_PREVIEW_PROFILER_ENABLED: "false"
     ENABLE_AUDIT_APPENDER: "TRUE"
 
+  externalConfigMapValues:
```
Comment (and-mora) [inline helm/values.yaml:64]: why externalConfigMapValues? do we have a config map defined in the infrastructure code?

**Review by and-mora — APPROVED** [review]

**Pull request merged**